src/pocketmine/level/WeakPosition.php

```diff
@@ -53,7 +53,18 @@ public function getLevel(){
 		return Server::getInstance()->getLevel($this->levelId);
 	}
 
-	public function setLevel(Level $level){
+	/**
+	 * @param Level|null $level
+	 *
+	 * @return $this
+	 *
+	 * @throws \InvalidArgumentException if the specified Level has been closed
+	 */
+	public function setLevel(Level $level = null){
+		if($level !== null and $level->isClosed()){
+			throw new \InvalidArgumentException("Specified level has been unloaded and cannot be used");
+		}
+
 		$this->levelId = ($level !== null ? $level->getId() : -1);
 		return $this;
```

Comment:
Not my doing. I left it alone since Position is very widely used and I don't fancy digging out the usages of this return value.